packages/shared/src/permissions/permissions.utils.ts

```diff
@@ -31,3 +31,77 @@ export function hasPermission(
 export function roleSupportsEnvLimit(role: MemberRole): boolean {
   return ["engineer", "experimenter"].includes(role);
 }
+
+export type ReadAccessFilter = {
+  globalReadAccess: boolean;
+  projects: { id: string; readAccess: boolean }[];
+};
+
+export function getReadAccessFilter(userPermissions: UserPermissions) {
+  const readAccess: ReadAccessFilter = {
+    globalReadAccess: userPermissions.global.permissions.readData || false,
+    projects: [],
+  };
+
+  Object.keys(userPermissions.projects).forEach((project) => {
+    readAccess.projects.push({
+      id: project,
+      readAccess:
+        userPermissions.projects[project].permissions.readData || false,
+    });
+  });
```

Comment:
Small code style recommendation - if you need both keys and values of an object, use `Object.entries()` instead of `Object.keys()`.

```suggestion
  Object.entries(userPermissions.projects).forEach(([project, projectPermissions]) => {
    readAccess.projects.push({
      id: project,
      readAccess: projectPermissions.permissions.readData || false,
    });
  });
```